pkg/test/framework/components/zipkin/kube.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkin
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sort"
+
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/resource"
+	testKube "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+)
+
+const (
+	appName    = "zipkin"
+	tracesAPI  = "/api/v2/traces"
+	zipkinPort = 9411
+)
+
+var (
+	_ Instance  = &kubeComponent{}
+	_ io.Closer = &kubeComponent{}
+)
+
+type kubeComponent struct {
+	id        resource.ID
+	address   string
+	forwarder testKube.PortForwarder
+	env       *kube.Environment
+}
+
+func newKube(ctx resource.Context) (Instance, error) {
+	env := ctx.Environment().(*kube.Environment)
+	c := &kubeComponent{
+		env: env,
+	}
+	c.id = ctx.TrackResource(c)
+
+	// Find the zipkin pod and service, and start forwarding a local port.
+	cfg, err := istio.DefaultConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	fetchFn := env.Accessor.NewSinglePodFetch(cfg.SystemNamespace, fmt.Sprintf("app=%s", appName))
```

Comment:
Is it possible to not assume that zipkins will be installed in `istio-system`? It won't be in https://github.com/istio/installer (1.2).

Reply:
I cannot find any tracer in https://github.com/istio/installer/tree/master/istio-telemetry. Is it to be added? If yes, what namespace is going to be used? istio-telemetry?

Reply:
We haven't added it yet, but will in the future. By default we will probably use `istio-telemetry`, but we want to support any namespaces. If that isn't feasible then its ok, we have a bunch of other tests that assume `istio-system` already

Reply:
I think this is fine as-is. I don't think it was decided which namespace zipkin should live (it can be telemetry or its own).

Reply:
I do wonder if the test should deploy zipkin... I think this will need to be discussed once the test framework switched to using the multi-namespace installer.

Reply:
@howardjohn for sure it is possible. To make it robust we need to get list of namespaces and try out every namespace to find the pod. I think this is kind of too much, so I prefer to leave it as it is for now.

@kyessenov Are you thinking about a fake zipkin backend?

Reply:
I think the new installer does not install zipkin (or at least doesn't have zipkin/jaeger by default). So the test should ensure zipkin is present and also configure the mesh config correctly. But that can be done later, after transitioning to the new installer.

Reply:
`kubectl get pods --all-namespaces -l app=prometheus` seems to work so I think we could do something like that. But we can change this another time, there are many instances to fix

Reply:
Ok thanks!